core/src/test/resources/org/apache/calcite/test/SqlHintsConverterTest.xml

```diff
@@ -55,6 +55,56 @@ select /*+ resource(cpu='2') */ avg(e2.sal) from emp e2 where e2.deptno = d1.dep
         </Resource>
         <Resource name="hints">
             <![CDATA[
+Project:[[RESOURCE inheritPath:[] options:{PARALLELISM=3}]]
+Project:[[RESOURCE inheritPath:[0] options:{PARALLELISM=3}]]
+Project:[[RESOURCE inheritPath:[] options:{CPU=2}], [RESOURCE inheritPath:[0, 0, 1, 0] options:{PARALLELISM=3}]]
+]]>
+        </Resource>
+    </TestCase>
+    <TestCase name="testHintsInSubQueryWithDecorrelation2">
+        <Resource name="sql">
+            <![CDATA[select /*+ properties(k1='v1', k2='v2'), index(ename), no_hash_join */
+sum(e1.empno) from emp e1, dept d1
+where e1.deptno = d1.deptno
+and e1.sal> (
+select /*+ properties(k1='v1', k2='v2'), index(ename), no_hash_join */
+  avg(e2.sal)
+  from emp e2
+  where e2.deptno = d1.deptno)]]>
+        </Resource>
+        <Resource name="hints">
+            <![CDATA[
+Project:[[PROPERTIES inheritPath:[] options:{K1=v1, K2=v2}], [INDEX inheritPath:[] options:[ENAME]], [NO_HASH_JOIN inheritPath:[]]]
```

Comment:
The following is rel plan of the testing sql. I noticed the size of hints for `Project 2` is 0. I'm wondering whether there is something wrong with hint propagation in decorrelation part.
```
LogicalAggregate(group=[{}], EXPR$0=[SUM($0)])
  1> LogicalProject(EMPNO=[$0])
    2> LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8], DEPTNO0=[$9], NAME=[$10], DEPTNO1=[CAST($11):INTEGER], EXPR$0=[CAST($12):INTEGER])
      LogicalJoin(condition=[AND(=($9, $11), >($5, $12))], joinType=[inner])
        LogicalJoin(condition=[=($7, $9)], joinType=[inner])
          LogicalTableScan(table=[[CATALOG, SALES, EMP]])
          LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
        LogicalAggregate(group=[{0}], EXPR$0=[AVG($1)])
          3> LogicalProject(DEPTNO=[$7], SAL=[$5])
            LogicalTableScan(table=[[CATALOG, SALES, EMP]])
```

Reply:
This is expected, because these 3 hints items are all not belong to the `PROJECT` node type, so they are all filtered out.

Reply:
Yeah, I just ignored the definition of hint strategy table...